### DISCUSSION TITLE
GPUI3

### DISCUSSION THREAD

**Discussion by mocenigo** [zed-industries/zed#21384] [Feature Requests]
This one is for the developers: @nathansobo and the others. What is in the plan for GPUI 3? I have seen there is a branch called GPUI3.

Also, how difficult would it be to implement a HTML canvas using GPUI?

**Comment by mocenigo** [comment]
I see no answer but I have another question: when I bump GPUI to 3.0 (there is a script to do that), Zed becomes way slower. I assume this is because of compilation in debug mode or debug information being generated. How to disable this?

**Reply by someone13574** [reply]
It is because you are building in debug mode. Also, the main ideas behind that branch ended up becoming this PR, which has been in main for a while now: https://github.com/zed-industries/zed/pull/22632

**Reply by mocenigo** [reply]
> It is also definitely because you are building in debug mode. Also, the main ideas behind that branch ended up becoming this PR, which has been in main for a while now: #22632

Thank you! I did not double check the build files because I was not building in debug mode before, and I did not ask the build script to. Weird, also I do not think that it is only GPUI to be built in debug mode. This should not happen. 

What are the advantages of GPUI 3.0 at this moment? Maybe @nathansobo can help here as well?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
